### PR TITLE
make oadm router and registry resiliant to missing client

### PR DESCRIPTION
**Pull request by deads2k**
`oc adm router --dry-run -o yaml` (and registry) shouldn't require an apiserver connection.  This makes it tolerant for cluster up

### CODE REVIEW

File: pkg/oc/admin/registry/registry.go [hunk 1]
```diff
@@ -283,19 +285,26 @@ func (opts *RegistryOptions) RunCmdRegistry() error {
...
+	case generate && opts.serviceClient == nil:
+	default:
+		service, err := opts.serviceClient.Services(opts.namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			if !generate {
+				if !errors.IsNotFound(err) {
+					return fmt.Errorf("can't check for existing docker-registry %q: %v", name, err)
+				}
+				if opts.Config.Action.DryRun {
+					return fmt.Errorf("Docker registry %q service does not exist", name)
```
Comment (mfojtik) [inline pkg/oc/admin/registry/registry.go]: nite: image registry

**Comment by deads2k** [comment]
Reworked to add a `--local` since code is relying on contacting the server when doing dry runs.

@mfojtik take another look?

**Comment by deads2k** [comment]
/assign @mfojtik 

prereq for cluster up

**Pull request merged**